pkg/agent/controller/endpoint.go

```diff
@@ -194,22 +194,24 @@ func (e *EndpointController) endPointDeleted(key string) error {
 }
 
 func (e *EndpointController) endpointSliceFromEndpoints(endpoints *corev1.Endpoints) *discovery.EndpointSlice {
-	endpointSlice := &discovery.EndpointSlice{}
 	controllerFlag := false
-	endpointSlice.Name = endpoints.Name + "-" + e.clusterID
-	endpointSlice.Labels = map[string]string{
-		lhconstants.LabelServiceImportName: e.serviceImportName,
-		discovery.LabelManagedBy:           lhconstants.LabelValueManagedBy,
-		lhconstants.LabelSourceNamespace:   e.serviceImportSourceNameSpace,
-		lhconstants.LabelSourceCluster:     e.clusterID,
-	}
-	endpointSlice.OwnerReferences = []metav1.OwnerReference{{
-		APIVersion:         "lighthouse.submariner.io.v2alpha1",
-		Kind:               "ServiceImport",
-		Name:               e.serviceImportName,
-		UID:                e.serviceImportUID,
-		Controller:         &controllerFlag,
-		BlockOwnerDeletion: nil,
+	endpointSlice := &discovery.EndpointSlice{ObjectMeta: metav1.ObjectMeta{
+		Name:      endpoints.Name + "-" + e.clusterID,
+		Namespace: e.serviceImportSourceNameSpace,
```

Comment:
Setting `Namespace` to that of the owner `ServiceImport` is the functional change here. The owned resource must be in the same namespace as the owner so this ensures that even tho it should be the same as the `Endpoints`.  

I inlined the rest of the code for improved readability.